src/NHibernate/IFutureValueAsync.cs

```diff
@@ -0,0 +1,9 @@
+using System.Threading.Tasks;
+
+namespace NHibernate
+{
+	public interface IFutureValueAsync<T>
+	{
+		Task<T> GetValue();
```

Comment:
> A single ToFuture returning an interface capable of both would be nice

One problem here is that both `IEnumerable` and `IAsyncEnumerable` have the same method `GetEnumerator` and also same extension methods like `ToList` so when the consumer would like to call `ToList` on the following interface:
```
public interface IFuture<T> : IEnumerable<T>, IAsyncEnumerable<T> { }
```
will get an ambiguous call that will prevent the code to be compiled
```
IFuture<Policy> future = GetFuture();
var list= future.ToList() // this will produce an ambiguous call error
```